src/figuresClasses.ts

```diff
@@ -1,19 +1,96 @@
 export interface Figure {
-
+  color: 'red' | 'green' | 'blue',
+  shape: 'triangle' | 'circle' | 'rectangle',
+  getArea(): number
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape = 'triangle';
 
-}
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const biggestSide: number = Math.max(a, b, c);
+    const error: string = `Sides ${a}, ${b}`
+    + `and ${c} can't form a triangle`;
+
+    switch (biggestSide) {
+      case a:
+        if ((b + c) <= a) {
+          throw new Error(error);
+        }
+        break;
+      case b:
+        if ((a + c) <= b) {
+          throw new Error(error);
+        }
+        break;
 
-export class Circle {
+      default:
+        if ((a + b) <= c) {
+          throw new Error(error);
+        }
+    }
 
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const s: number = (this.a + this.b + this.c) / 2;
+
+    return +Math.sqrt(s * ((s - this.a) * (s - this.b) * (s - this.c)))
+      .toFixed(2);
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape = 'circle';
+
+  constructor(
+    public color: string,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius is not valid');
+    }
```

Comment:
you need to check input arguments for each figure
let's create separate function for this purpose, that will accept needed params, and will check that each of them is bigger than 0
and will call this function in each figure